### DISCUSSION TITLE
(0,0): Unexpected "Identifier" in class body. / p_line -1 is out of bounds

### DISCUSSION THREAD

**Discussion by grayocene** [ephread/inkgd#88] [Q&A] [answered]
Hi, I'm using the sample code in the documentation, but I keep on getting a (0,0): Unexpected "Identifier" in class body error, and a p_line = -1 is out of bounds error. 

The output errors I get are:
 - res://ink_template.gd:0 - Parse Error: Unexpected "Identifier" in class body.
 - modules/gdscript/gdscript.cpp:2936 - Failed to load script "res://ink_template.gd" with error "Parse error". (User)
 - scene/gui/code_edit.cpp:1655 - Index p_line = -1 is out of bounds (get_line_count() = 37).
 - scene/gui/text_edit.cpp:6214 - Index p_line = -1 is out of bounds (text.size() = 37).
 - scene/gui/text_edit.cpp:6208 - Index p_line = -1 is out of bounds (text.size() = 37).

sample code:
```
onready var _ink_player = $InkPlayer


func _ready():
	_ink_player.connect("loaded", self, "_story_loaded")
	_ink_player.create_story()


func _story_loaded(successfully: bool):
	if !successfully:
		return

	_continue_story()


func _continue_story():
	while _ink_player.can_continue:
		var text = _ink_player.continue_story()

		# This text is a line of text from the ink story.
		# Set the text of a Label to this value to display it in your game.
		print(text)

	if _ink_player.has_choices:
		# 'current_choices' contains a list of the choices, as strings.
		for choice in _ink_player.current_choices:
			print(choice)

		# '_select_choice' is a function that will take the index of
		# your selection and continue the story by calling again
		# `_continue_story()`.
		_select_choice(0)

	else:
		# This code runs when the story reaches it's end.
		print("The End")
```

Any help would be appreciated. I've made sure to enable the global variable __InkRuntime, if that's a factor.

![inkgdruntimeerror](https://github.com/user-attachments/assets/f6a99eb2-c01f-4c70-b5c5-353940729397)

**Comment by francoisdlt** [comment, marked as answer]
this sample code seems to be for Godot 3
make sure to adapt it for Godot 4

**Reply by francoisdlt** [reply]
are you sure you are using the `godot4` branch of this plugin ?

**Reply by grayocene** [reply]
Oh, I see. Silly mistake, my bad. I'm new to godot, if you couldn't tell LOL

I am using the godot4 branch, just was using that godot 3 code. Do you know if there's any sample code that's in godot 4?
